src/reducers/query.ts

```diff
@@ -87,16 +87,32 @@ export default {
   query: handleActions(
     {
       [ActionTypes.PREVIEW_DATA_SUCCESS]: (state, action: any) => {
-        const rows = action.payload.queryResults.data.result;
         const columnsByName = _.keyBy(state.columns, 'name');
-        const columns = action.payload.columns.map((column: ColumnModel) => ({
+        const columns: TableColumnType[] = action.payload.columns.map((column: ColumnModel) => ({
           name: column.name,
           dataType: column.datatype,
           arrayOf: column.array_of,
           allowSort: !column.array_of,
           allowResize: true,
           width: columnsByName[column.name]?.width || TABLE_DEFAULT_COLUMN_WIDTH,
         }));
+        // We only need to re-format row data of type timestamp
+        const timestampColumns: TableColumnType[] = [];
+        columns.forEach((col: TableColumnType) => {
+          if (col.dataType === 'timestamp') {
+            timestampColumns.push(col);
+          }
+        });
+        const rows = action.payload.queryResults.data.result.map((row: any) => {
+          timestampColumns.forEach((col: TableColumnType) => {
+            if (col.arrayOf) {
+              row[col.name] = row[col.name].map((v: number) => new Date(v * 1000).toLocaleString());
+            } else {
+              row[col.name] = new Date(row[col.name] * 1000).toLocaleString();
```

Comment:
Is there a way to break out this formatting so that it can be reused between the array and singleton cases?